src/coreclr/jit/hwintrinsicarm64.cpp

```diff
@@ -709,6 +709,30 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
+        case NI_Vector64_AddSaturate:
+        case NI_Vector128_AddSaturate:
+        {
+            assert(sig->numArgs == 2);
+
+            if (simdSize == 8 && varTypeIsLong(simdBaseType))
+            {
+                break;
+            }
```

Comment:
This shouldn't be skipped for `TYP_LONG`, it should just use `AddSaturateScalar` which is already exposed.